models/classes/Platform/Repository/Lti1p3RegistrationSnapshotRepository.php

```diff
@@ -0,0 +1,232 @@
+                    'oidc_authentication_url' => $ltiPlatformRegistration->getOidcAuthenticationUrl(),
+                    'oauth2_access_token_url' => $ltiPlatformRegistration->getOAuth2AccessTokenUrl(),
+                    'jwks_url' => $ltiPlatformRegistration->getJwksUrl(),
+                    'updated_at' => date('Y-m-d H:i:s')
+                ]
+            );
+
+            return;
+        }
+
+        $this->getPersistence()->updateMultiple(
+            'lti_platform_registration',
+            [
+                [
+                    'conditions' => [
+                        'statement_id'=> $ltiPlatformRegistration->getIdentifier()
+                    ],
+                    'updateValues' => [
+                        'name' => $ltiPlatformRegistration->getName(),
+                        'audience' => $ltiPlatformRegistration->getAudience(),
+                        'client_id' => $ltiPlatformRegistration->getClientId(),
+                        'deployment_id' => $ltiPlatformRegistration->getDeploymentId(),
+                        'oidc_authentication_url' => $ltiPlatformRegistration->getOidcAuthenticationUrl(),
+                        'oauth2_access_token_url' => $ltiPlatformRegistration->getOAuth2AccessTokenUrl(),
+                        'jwks_url' => $ltiPlatformRegistration->getJwksUrl(),
+                        'updated_at' => date('Y-m-d H:i:s')
+                    ]
+                ]
+            ]
+        );
+    }
+
+    public function find(string $identifier): ?RegistrationInterface
+    {
+        $row = $this->getRow(['statement_id' => $identifier]);
+
+        if (empty($row)) {
+            return null;
+        }
+
+        return $this->toRegistration($row);
+    }
+
+    public function findAll(): array
+    {
+        $rows = $this->getPersistence()
+            ->query('SELECT * FROM lti_platform_registration')
+            ->fetchAll();
+
+        return array_map(
+            function (array $row) {
+                return $this->toRegistration($row);
+            },
+            $rows
+        );
+    }
+
+    public function findByClientId(string $clientId): ?RegistrationInterface
+    {
+        $row = $this->getRow(['client_id' => $clientId]);
+
+        if (empty($row)) {
+            return null;
+        }
+
+        return $this->toRegistration($row);
+    }
+
+    public function findByPlatformIssuer(string $issuer, string $clientId = null): ?RegistrationInterface
+    {
+        $queryParams = [
+            'audience' => $issuer,
+            'client_id' => $clientId
+        ];
+
+        $row = $this->getRow($queryParams);
+
+        if (empty($row)) {
+            return null;
+        }
+
+        return $this->toRegistration($row);
+    }
+
+    public function findByToolIssuer(string $issuer, string $clientId = null): ?RegistrationInterface
+    {
+    }
+
+    private function getRow(array $queryParams = []): array
```

Comment:
WDYT to move place this `private` method after  `public`  `deleteByStatementId`  ?

Reply:
Wow! Absolutely, thanks! ;)